model_bakery/baker.py

```diff
@@ -319,7 +319,11 @@ def prepare(self, _save_related=False, **attrs: Any) -> Model:
         return self._make(commit=False, commit_related=_save_related, **attrs)
 
     def get_fields(self) -> Any:
-        return self.model._meta.fields + self.model._meta.many_to_many
+        return (
+            self.model._meta.fields
+            + self.model._meta.many_to_many
+            + tuple(self.model._meta.private_fields)
+        )
```

Comment:
@berinhard Any reason we can't use [`self.model._meta.get_fields()`](https://docs.djangoproject.com/en/3.2/ref/models/meta/) here and then remove `related_objects` via `set` subtraction (as they are retrieved via another method already, `get_related`)?

```suggestion
        return set(self.model._meta.get_fields()) - set(self.get_related())
```

The results may be more complex to read, but it would be future-proof going forward.

Reply:
Yes, I think using Django's public API in _meta over private API is preferable too. The `get_related` method itself could also be updated to only use public API methods (perhaps also other areas in the Baker class), but that seems outside the scope of this PR at least. I've updated this method according to the suggestion.